test/runAll.unmocked.spec.js

```diff
@@ -1,12 +1,244 @@
+    await appendFile('test.js', testJsFileUnfixable)
+    await execGit(['add', 'test.js'])
+    const status = await execGit(['status'])
+
+    // Run lint-staged with `prettier --write` to break the linter
+    const success = await gitCommit({ config: { '*.js': ['prettier --write', 'git add'] } })
+    expect(success).toEqual(false)
+
+    // Something was wrong so the repo is returned to original state
+    expect(await execGit(['rev-list', '--count', 'HEAD'])).toEqual('1')
+    expect(await execGit(['log', '-1', '--pretty=%B'])).toMatch('initial commit')
+    expect(await execGit(['status'])).toEqual(status)
+    expect(await readFile('test.js')).toEqual(testJsFileUnfixable)
+  })
+
+  it('Should commit partial change from partially staged file when no errors from linter', async () => {
+    // Stage pretty file
+    await appendFile('test.js', testJsFilePretty)
+    await execGit(['add', 'test.js'])
+
+    // Edit pretty file but do not stage changes
+    const appended = '\nconsole.log("test");\n'
+    await appendFile('test.js', appended)
+
+    // Run lint-staged with `prettier --list-different` and commit pretty file
+    const success = await gitCommit({ config: { '*.js': 'prettier --list-different' } })
+    expect(success).toEqual(true)
+
+    // Nothing is wrong, so a new commit is created and file is pretty
+    expect(await execGit(['rev-list', '--count', 'HEAD'])).toEqual('2')
+    expect(await execGit(['log', '-1', '--pretty=%B'])).toMatch('test')
+
+    // Latest commit contains pretty file
+    // `git show` strips empty line from here here
+    expect(await execGit(['show', 'HEAD:test.js'])).toMatch(testJsFilePretty.replace(/\n$/, ''))
+
+    // Since edit was not staged, the file is still modified
+    expect(await execGit(['status'])).toMatch('modified:   test.js')
+    expect(await readFile('test.js')).toMatch(testJsFilePretty + appended)
+  })
+
+  it('Should commit partial change from partially staged file when no errors from linter and linter modifies file', async () => {
+    // Stage ugly file
+    await appendFile('test.js', testJsFileUgly)
+    await execGit(['add', 'test.js'])
+
+    // Edit ugly file but do not stage changes
+    const appended = '\n\nconsole.log("test");\n'
+    await appendFile('test.js', appended)
+
+    // Run lint-staged with `prettier --write` and commit pretty file
+    const success = await gitCommit({ config: { '*.js': ['prettier --write', 'git add'] } })
+    expect(success).toEqual(true)
+
+    // Nothing is wrong, so a new commit is created and file is pretty
+    expect(await execGit(['rev-list', '--count', 'HEAD'])).toEqual('2')
+    expect(await execGit(['log', '-1', '--pretty=%B'])).toMatch('test')
+
+    // Latest commit contains pretty file
+    // `git show` strips empty line from here here
+    expect(await execGit(['show', 'HEAD:test.js'])).toMatch(testJsFilePretty.replace(/\n$/, ''))
+
+    // Nothing is staged
+    expect(await execGit(['status'])).toMatch(`On branch master
```

Comment:
Consider using Jest `toMatchInlineSnapshot` for such tests since it’s easier to update and maintain

Reply:
Sure thing, I’ll make changes.